lm_eval/models/huggingface.py

```diff
@@ -147,17 +149,23 @@ def __init__(
             assert isinstance(pretrained, str)
             assert isinstance(batch_size, (int, str))
 
-            gpus = torch.cuda.device_count()
+            device_counts = torch.cuda.device_count()
+
             accelerator_kwargs = InitProcessGroupKwargs(timeout=timedelta(weeks=52))
             accelerator = Accelerator(kwargs_handlers=[accelerator_kwargs])
+
             if accelerator.num_processes > 1:
                 self.accelerator = accelerator
 
+            if is_npu_available():
+                device_counts = torch.npu.device_count()
```

Comment:
What if there are some systems with different counts of cuda devices and npus at the same time, available to users?

Reply:
Hi, thanks for your advice. I just double check that cuda devices and npus could not be set in same matchine simultaneously.

Besides, I think it's inflexible to set `gpus` and `npus` variables together, which will brings more conditional statements in following code.

Best